homeassistant/components/onewire/sensor.py

```diff
@@ -100,19 +217,25 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class OneWire(Entity):
     """Implementation of an One wire Sensor."""
 
-    def __init__(self, name, device_file, sensor_type):
+    def __init__(self, name, device_file, sensor_type, owproxy=None):
         """Initialize the sensor."""
         self._name = name + " " + sensor_type.capitalize()
         self._device_file = device_file
         self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
         self._state = None
+        self._owproxy = owproxy
 
     def _read_value_raw(self):
         """Read the value as it is returned by the sensor."""
         with open(self._device_file, "r") as ds_device_file:
             lines = ds_device_file.readlines()
         return lines
 
+    def _read_value_ownet(self):
+        """Read a value from the owserver."""
+        if self._owproxy:
+            return bytes.decode(self._owproxy.read(self._device_file)).lstrip()
+
```

Comment:
We need to explicitly return something in this case too, ie `return None`. If we `return` explicitly anywhere in a function we should return explicitly in all cases.